examples/akka-http-service/src/main/scala/com/wookiee/basic/AkkaHttpService.scala

```diff
@@ -1,10 +1,136 @@
+import akka.http.scaladsl.model.{HttpMethods, StatusCodes}
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.Route
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import com.webtrends.harness.component.akkahttp.routes.{AkkaHttpEndpointRegistration, AkkaHttpRequest, EndpointType, RouteGenerator}
+import com.webtrends.harness.logging.Logger
 import com.webtrends.harness.service.Service
-import com.wookiee.basic.command.AkkaHttpReadCommand
+import org.json4s.{DefaultFormats, Formats}
+import org.json4s.ext.JodaTimeSerializers
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+case class Message(message: String)
+case class NotAuthorized(message: String) extends Exception
+case class Forbidden(message: String) extends Exception
+
+class AkkaHttpService extends Service with AkkaHttpEndpointRegistration {
+  implicit val timeout: Timeout = Timeout(2.seconds)
+  def formats: Formats = DefaultFormats ++ JodaTimeSerializers.all
+  implicit val sys: ActorSystem = context.system
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
 
-class AkkaHttpService extends Service {
   override def addCommands: Unit = {
-    addCommand(AkkaHttpReadCommand.CommandName, classOf[AkkaHttpReadCommand])
+    // Get endpoint
+    addAkkaHttpEndpoint[Message, Message](
+      "simple.get",
+      "getTest",
+      HttpMethods.GET,
+      EndpointType.INTERNAL,
+      reqToHello,
+      echo[Message],
+      stringResponse,
+      authorizationRejections,
+      _ => "myId"
+    )
+
+    // POST endpoint
+    addAkkaHttpEndpoint[Message, Message](
+      "post.test",
+      "postTest",
+      HttpMethods.POST,
+      EndpointType.INTERNAL,
+      reqToPayloadMessage,
+      echo[Message],
+      stringResponse,
+      authorizationRejections
+    )
+
+    // GET endpoint with segments
+    addAkkaHttpEndpoint[List[String], List[String]](
+      "segment.lookup",
+      "getTest/$name",
+      HttpMethods.GET,
+      EndpointType.INTERNAL,
+      reqToRouteParams,
+      echo[List[String]],
+      paramsResponse,
+      authorizationRejections
+    )
+
+    addAkkaHttpEndpoint[List[String], List[String]](
+      "report.lookup",
+      "account/$accountGuid/report/$reportId",
+      HttpMethods.GET,
+      EndpointType.INTERNAL,
+      reqToRouteParams,
+      echo[List[String]],
+      paramsResponse,
+      authorizationRejections
+    )
+
+    addAkkaHttpEndpoint[List[String], List[String]](
+      "report.lookup.forbidden",
+      "report/$reportId",
+      HttpMethods.GET,
+      EndpointType.INTERNAL,
+      rejectReport1Calls,
+      echo[List[String]],
+      paramsResponse,
+      authorizationRejections
+    )
+  }
+
+  def reqToHello(r: AkkaHttpRequest): Future[Message] = Future.successful(Message("Hello World"))
+  def reqToPayloadMessage(r: AkkaHttpRequest): Future[Message] =
+    RouteGenerator.entityToString(r.requestBody.get).map(Message)
```

Comment:
No it is not the right place, forgot to pull that out to some new module. Honestly it's not super complex so it might be ok to remove and just expect consumers to implement themselves.